tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -295,6 +297,20 @@ TangoErrorType TangoRosNode::OnTangoServiceConnected() {
   return TANGO_SUCCESS;
 }
 
+void TangoRosNode::RequestADFPermissionAndWaitForAnswer() {
+  UpdateAndPublishTangoStatus(TangoStatus::NEED_TO_REQUEST_ADF_PERMISSION);
+  while (tango_status_ != TangoStatus::ADF_PERMISSION_REQUEST_ANSWERED) {
+    ros::Duration(0.1).sleep();
+  }
```

Comment:
Is abusing the /tango/status the best way to do this? Why not using a ros service call between the java side (server) and the c++ side (client)?